src/cascadia/TerminalControl/HwndTerminal.hpp

```diff
@@ -106,6 +122,9 @@ struct HwndTerminal : ::Microsoft::Console::Types::IControlAccessibilityInfo
     std::optional<til::point> _lastMouseClickPos;
     std::optional<til::point> _singleClickTouchdownPos;
 
+    Microsoft::Console::TSF::Handle _tsfHandle;
+    TsfDataProvider _tsfDataProvider;

```

Comment:
The destruction order is wrong here. You need to destroy the `Handle` first, since the `TsfDataProvider` doesn't properly implement `AddRef`/`Release`. I think this is worth commenting as well.